jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleLayerConfigurations.java

```diff
@@ -49,13 +51,17 @@
    * @throws IOException if an I/O exception occurred during resolution
    */
   static JavaLayerConfigurations getForProject(
-      Project project, Logger logger, Path extraDirectory, AbsoluteUnixPath appRoot)
+      Project project,
+      Logger logger,
+      Path extraDirectory,
+      Map<AbsoluteUnixPath, FilePermissions> permissions,
```

Comment:
Javadoc about `permissions`. Also, it should be obvious from name or Javadoc that these file permissions apply only to the files from `extraDirectory`.